### PR TITLE
Episode 3 improvements

### LINKED ISSUE #253: Improvements for episode 3 from teaching
@dafnevk made some suggestions for improvements after teaching episode 3:
- Variable naming is not ideal 
- Simplify the code. There's a lot of plotting  boilerplate, but also unnecessary options like verbose, batch_size, etc 
- Put compile model, and plotting results into function? Have to do a lot of copypasting now 
- I skipped the ‘try out metrics’ exercise, didn't seem necessary 

We can use [dafne's notebook](https://github.com/esciencecenter-digital-skills/2022-09-26-ds-dl-intro/blob/main/files/03-monitor-the-model.ipynb) for inspiration

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #253 : 
* Removes exercise on metrics
* Adds 3 functions to remove boilerplate code:
    - `compile_model`
    - `plot_history`
    - `plot_predictions`
* Simplifies the code in a few places, for example:
   - remove `verbose` arguments to `model.fit`

### CODE REVIEW

File: _episodes/03-monitor-the-model.md [hunk 1]
```diff
@@ -356,27 +358,31 @@ For the present *regression task*, it makes more sense to compare true and predi
...
 
 ~~~
-fig, axes = plt.subplots(1, 2, figsize=(12, 6))
-plt.style.use('ggplot')  # optional, that's only to define a visual style
-axes[0].scatter(y_train_predicted, y_train, s=10, alpha=0.5, color="teal")
-axes[0].set_title("training set")
-axes[0].set_xlabel("predicted sunshine hours")
-axes[0].set_ylabel("true sunshine hours")
+# We define a function that we will reuse in this lesson
+def plot_predictions(y_pred, y_true, title):
```
Comment (CunliangGeng) [inline _episodes/03-monitor-the-model.md:362]: It's more intuitive to use actual values as x, predicted as y. In this way one could easily see if the predicted is larger than actual value by checking if the position of predicted value is higher than diagonal or not. "Higher is larger" is more intuitive than "more right is larger" :-)

File: _episodes/03-monitor-the-model.md [hunk 2]
```diff
@@ -322,15 +314,25 @@ history = model.fit(X_train, y_train,
...
-history_df = pd.DataFrame.from_dict(history.history)
-sns.lineplot(data=history_df['root_mean_squared_error'])
-plt.xlabel("epochs")
-plt.ylabel("RMSE")
+def plot_history(metrics):
+    """
+    Plot the training history
+
+    Args:
+        metrics: A list of metrics to plot
```
Comment (CunliangGeng) [inline _episodes/03-monitor-the-model.md:328]: Change it to `Metrics(str, list): Metric or a list of metrics to plot`

File: _episodes/03-monitor-the-model.md [hunk 3]
```diff
@@ -322,15 +314,25 @@ history = model.fit(X_train, y_train,
...
+    """
+    history_df = pd.DataFrame.from_dict(history.history)
+    sns.lineplot(data=history_df[metrics])
+    plt.xlabel("epochs")
+    plt.ylabel("RMSE")
+
+plot_history(['root_mean_squared_error'])
```
Comment (CunliangGeng) [inline _episodes/03-monitor-the-model.md:335]: so you can directly use `plot_history('root_mean_squared_error')`

**Review by CunliangGeng — COMMENTED** [review]
Thanks, most of the changes look good.

**Review by dsmits — APPROVED** [review]

File: _episodes/03-monitor-the-model.md (hunk 1 above)
Comment (dsmits) [reply _episodes/03-monitor-the-model.md:362]: Sounds good to me

File: _episodes/03-monitor-the-model.md (hunk 2 above)
Comment (dsmits) [reply _episodes/03-monitor-the-model.md:328]: Is that a convention for docstrings? Which style?

Comment (CunliangGeng) [reply _episodes/03-monitor-the-model.md:328]: google style https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: _episodes/03-monitor-the-model.md (hunk 1 above)
Comment (svenvanderburg) [reply _episodes/03-monitor-the-model.md:362]: I think both are used in the field. But I get your reasoning. I created an issue for it: #289
